### PR TITLE
Feature/mariadb integration ci

### LINKED ISSUE #238: CI: Add MariaDB support (i.e. let integration tests run using MariaDB as a server)
##### SUMMARY

As there are issues specific to MariaDB reported and our fixes / features can affect MariaDB users, we need to have MariaDB in CI

Related files:
- .github/workflows/ansible-test-plugins.yml
- tests/integration/targets/setup_mysql/*

There's also currently not used role https://github.com/ansible-collections/community.mysql/tree/main/tests/integration/old_mariadb_replication that were used when the collection's content lived in ansible/ansible (might be helpful)

##### ISSUE TYPE
- Feature Idea

### PR DESCRIPTION
**Pull request by rsicart**
##### SUMMARY
CI: Add MariaDB support

Fixes #238

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
all

### CODE REVIEW

**Comment by rsicart** [comment]
Do not hesitate to contribute to this PR :)

**Comment by Jorge-Rodriguez** [comment]
@rsicart Brilliant! Thank you!

**Comment by Jorge-Rodriguez** [comment]
@rsicart Please see discussion #141, in particular [this comment](https://github.com/ansible-collections/community.mysql/discussions/141#discussioncomment-643657)

**Comment by Andersson007** [comment]
@rsicart this is a great undertaking, thanks!

**Comment by Andersson007** [comment]
As some tests with MariaDB have failed
```
ERROR: The 4 integration test(s) listed below (out of 7) failed. See error output above for details:
test_mysql_info
test_mysql_replication
test_mysql_role
test_mysql_user
```
if the failings are not related to CI implementation introduced by this PR, we could try to skip them for now (e.g. if we tests mysql_info and it fails because of module implementation or something else not related to this PR, skip it when MariaDB).
We could create an issue, something like "Fix and enable testing with MariaDB for ..., ..., modules".
Then we could fix them one by one with separate PRs.
At least `mysql_role` failings are unrelated and, I think, all of the others as well.
I wouldn't like to see @rsicart stuck with unrelated things.

**Comment by rsicart** [comment]
The commit **Do not test mysql_replication_channel.yml on mariadb** does not test replication channel on mariadb.
That will need opening an Issue to test it on mariadb 10.7 (support to channel added on this version).
And perhaps when mariadb < 10.7, doing the same tests in _mysql_replication_channel.yaml_ but replacing _channel_ by _connection_name_.

**Comment by Andersson007** [comment]
> The commit **Do not test mysql_replication_channel.yml on mariadb** does not test replication channel on mariadb. That will need opening an Issue to test it on mariadb 10.7 (support to channel added on this version). And perhaps when mariadb < 10.7, doing the same tests in _mysql_replication_channel.yaml_ but replacing _channel_ by _connection_name_.

Feel free to exclude it and create the issue. As i suggested, feel free to exclude all things from testing against MariaDB when it seems to be too time / effort consuming to fix in this PR (creating issues to fix them separately later).

**Comment by Andersson007** [comment]
```
ERROR: The 2 integration test(s) listed below (out of 7) failed. See error output above for details:
test_mysql_role
test_mysql_user
```
Great progress! Thanks!

**Comment by rsicart** [comment]
I wrote some FIXME comments in order to find easily the issues to open :)

**Comment by Andersson007** [comment]
@rsicart great job! Thanks!
I'll take a look on Monday morning.
We should not forget to create an issue (after this PR is merged) describing what are disabled pointing to this PR.
If @Jorge-Rodriguez also takes a look, it would be great.
Thank you! (super important work that covers a huge gap in CI and, resp., in stability of the project for MariaDB folks)

File: changelogs/fragments/246-mariadb-integration-ci.yml
Comment (Andersson007) [inline changelogs/fragments/246-mariadb-integration-ci.yml]:
```suggestion
  - CI - add ci tests for mariadb (https://github.com/ansible-collections/community.mysql/issues/238 https://github.com/ansible-collections/community.mysql/pull/246).
```
The changelog fragment is not needed in this case as well as for all other changes that don't touch the code like doc ones for example

**Comment by Andersson007** [comment]
closed-reopened to relaunch tests

**Comment by Andersson007** [comment]
(ah, got it (`[ci skip]`), relaunched for the prev commit)

**Review by Andersson007 — APPROVED** [review]
LGTM
Let's wait for feedback from @bmalynovytch and @Jorge-Rodriguez until tomorrow, if there are no objections, we'll merge the PR. Thank you!

**Comment by Andersson007** [comment]
Ah, also we should remove the fragment. In the other parts, LGTM

**Review by bmalynovytch — APPROVED** [review]
LGTM 👍

File: changelogs/fragments/246-mariadb-integration-ci.yml
Comment (Andersson007) [reply changelogs/fragments/246-mariadb-integration-ci.yml]: @rsicart could you please remove the fragment? Then we can merge the PR

**Comment by Andersson007** [comment]
@rsicart thanks for the great CI improvement!
@bmalynovytch thanks for reviewing!

@rsicart feel free to create the issue

**Pull request merged**
